docs/pages/management/dynamic-configuration.mdx

```diff
@@ -0,0 +1,272 @@
+reference](../reference/cli.mdx#tctl) for more information on `tctl`.
+
+### Teleport Terraform provider
+
+Teleport's Terraform provider lets you manage your Teleport configuration
+resources within the same infrastructure-as-code source as the rest of your
+infrastructure. There is a Terraform resource for each Teleport configuration
+resource. For example:
+
+```text
+resource "teleport_role" "developer" {
+  metadata = {
+    name        = "developer"
+  }
+
+  spec = {
+    allow = {
+      logins = ["ubuntu", "debian", "{{internal.logins}}"]
+
+      node_labels = {
+        key   = ["env"]
+        value = ["test"]
+      }
+    }
+  }
+}
+```
+
+[Get started with the Terraform provider](./dynamic-configuration/terraform-provider.mdx).
+
+### Teleport Kubernetes Operator
+
+The Teleport Kubernetes Operator lets you apply Teleport configuration resources
+as Kubernetes resources so you can manage your Teleport settings alongside the
+rest of your Kubernetes infrastructure. Here is an example of a `TeleportRole`
+resource, which is equivalent to the two roles shown above:
+
+```yaml
+
+apiVersion: resources.teleport.dev/v5
+kind: TeleportRole
+metadata:
+  name: developer
+spec:
+  allow:
+    logins: ['ubuntu', 'debian', '{{internal.logins}}']
+    node_labels:
+      'env': 'test'
+```
+
+[Get started with the Kubernetes Operator](dynamic-configuration/teleport-operator.mdx). 
+
+## Reconciling the configuration file with dynamic resources
+
+Some dynamic configuration resources assign the same settings as fields within
+Teleport's static configuration file. For these fields, the Teleport Auth
+Service reconciles static and dynamic configurations on startup and when you
+create or remove a configuration resource.
+
+### Configuration resources that apply to static configuration fields
+
+There are three dynamic configuration resources that share fields with the
+static configuration file:
+
+- `session_recording_config`
+- `cluster_auth_preference`
+- `cluster_networking_config`
+
+#### `session_recording_config`
+
+|Dynamic resource field|Static configuration field|
+|---|---|
+|`spec.mode`|`auth_service.session_recording`|
+|`spec.proxy_checks_host_keys`|`auth_service.proxy_checks_host_keys`|
+
+#### `cluster_auth_preference`
+
+|Dynamic resource field|Static configuration field|
+|---|---|
+|`spec.type`|`auth_service.authentication.type`|
+|`spec.second_factor`|`auth_service.authentication.second_factor`|
+|`spec.connector_name`|`auth_service.authentication.connector_name`
+|`spec.u2f`|`auth_service.authentication.u2f`|
+|`spec.disconnect_expired_cert`|`auth_service.disconnect_expired_cert`|
+|`spec.allow_local_auth`|`auth_service.authentication.local_auth`|
+|`spec.message_of_the_day`|`auth_service.message_of_the_day`|
+|`spec.locking_mode`|`auth_service.authentication.locking_mode`|
+|`spec.webauthn`|`auth_service.authentication.webauthn`|
+|`spec.require_session_mfa`|`auth_service.authentication.require_session_mfa`|
```

Comment:
To avoid breaking keys across lines:

```suggestion
|<nobr>`spec.disconnect_expired_cert`</nobr>|`auth_service.disconnect_expired_cert`|
|`spec.allow_local_auth`|`auth_service.authentication.local_auth`|
|`spec.message_of_the_day`|`auth_service.message_of_the_day`|
|`spec.locking_mode`|`auth_service.authentication.locking_mode`|
|`spec.webauthn`|`auth_service.authentication.webauthn`|
|`spec.require_session_mfa`|<nobr>`auth_service.authentication.require_session_mfa`</nobr>|
```